src/index.ts

```diff
@@ -17,16 +17,17 @@ function getDimensionObject(node: HTMLElement): DimensionObject {
 }
 
 function useDimensions({
-    liveMeasure = true
+    liveMeasure = true,
 }: UseDimensionsArgs = {}): UseDimensionsHook {
     const [dimensions, setDimensions] = useState({});
     const [node, setNode] = useState(null);
-
+    const useLayoutHookBasedOnEnvironment =  typeof window === undefined ? useEffect : useLayoutEffect;
```

Comment:
I just wanted to let you know this code won't work, because typeof will always return a string. This works:

`typeof window === "undefined" [...]`

Reply:
Oh. My bad. Fixed it. Thanks

Reply:
Agree this would be nice to get merged.

Reply:
+1